AsyncDisplayKit/Private/ASMultidimensionalArrayUtils.mm

```diff
@@ -142,6 +164,18 @@ void ASDeleteElementsInMultidimensionalArrayAtIndexPaths(NSMutableArray *mutable
   return res;
 }
 
+NSArray<NSIndexPath *> *ASIndexPathsInMultidimensionalArrayIntersectingIndexPaths(NSArray *multidimensionalArray, NSArray<NSIndexPath *> *indexPaths)
+{
+  NSMutableArray *res = [NSMutableArray array];
+  for (NSIndexPath *indexPath in indexPaths) {
+    if (ASElementExistsAtIndexPathForMultidimensionalArray(multidimensionalArray, indexPath)) {
+      [res addObject:indexPath];
+    }
+  }
+
+  return res;
```

Comment:
Returning mutable arrays is the precedent set by the code in this file—I wasn't sure if I should change it as part of this